api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -35,6 +35,15 @@
  */
 public interface Catalog {
 
+  /**
+   * Return the name for this catalog.
+   *
+   * @return this catalog's name
+   */
+  default String name() {
```

Comment:
I did not intend to add the name to the `Catalog` API but we need this for proper metadata table names.